### PR TITLE
Flav/fix assistant builder close

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR changes the behavior when closing the the assistant builder. It uses the router to redirect the user where they were before entering the assistant builder.

Tested on both personal assistants and workspace assistants flows.

### CODE REVIEW

File: front/pages/w/[wId]/builder/assistants/index.tsx [hunk 1]
```diff
@@ -209,7 +209,9 @@ export default function WorkspaceAssistants({
                 />
               </Link>
               {workspaceAgents.length > 0 && (
-                <Link href={`/w/${owner.sId}/builder/assistants/new`}>
+                <Link
+                  href={`/w/${owner.sId}/builder/assistants/new?flow=workspace_assistants`}
```
Comment (flvndvd) [inline front/pages/w/[wId]/builder/assistants/index.tsx:213]: 🚗

**Comment by flvndvd** [comment]
FYI @spolu since you changed did change some lines of code around this part yesterday in https://github.com/dust-tt/dust/pull/3607.

**Review by spolu — APPROVED** [review]
LGTM

Only thing is that it will behave weirdly if the user is linked to that page directly (which is currently the only way to link to an assistant in Dust)?

**Comment by flvndvd** [comment]
> directly

I'm not sure to get the point. With the new refactoring all the paths that open this "full screen modal" are using `<Link>` which automatically pushes new URLs in the router, ensuring that we can properly use the router to bring the users back where they were. The only issue could be if you click on a link that opens a new tab and brings you right on this page, in which case when you close we will bring you to the empty tab. I'll find a workaround for this specific case.

**Comment by flvndvd** [comment]
> > directly
> 
> I'm not sure to get the point. With the new refactoring all the paths that open this "full screen modal" are using `<Link>` which automatically pushes new URLs in the router, ensuring that we can properly use the router to bring the users back where they were. The only issue could be if you click on a link that opens a new tab and brings you right on this page, in which case when you close we will bring you to the empty tab. I'll find a workaround for this specific case.

Ok so the history API is pretty limited and does not offer any easy way to check if the previous page belonged to our application. A bigger refactoring is needed to fully support the router to handle all router updates in our app. One temporary solution, pretty easy to implement could be to track just for this case the previous page by adding it in the URL (like we do with flow). It's pretty ugly, but very easy to do and given the pain of today's interaction with the new edition menu, IMHO it's worth a shot.

**Comment by flvndvd** [comment]
> > > directly
> > 
> > 
> > I'm not sure to get the point. With the new refactoring all the paths that open this "full screen modal" are using `<Link>` which automatically pushes new URLs in the router, ensuring that we can properly use the router to bring the users back where they were. The only issue could be if you click on a link that opens a new tab and brings you right on this page, in which case when you close we will bring you to the empty tab. I'll find a workaround for this specific case.
> 
> Ok so the history API is pretty limited and does not offer any easy way to check if the previous page belonged to our application. A bigger refactoring is needed to fully support the router to handle all router updates in our app. One temporary solution, pretty easy to implement could be to track just for this case the previous page by adding it in the URL (like we do with flow). It's pretty ugly, but very easy to do and given the pain of today's interaction with the new edition menu, IMHO it's worth a shot.

Updated with a temporary hack until I fix the internal router early next week.

**Review by philipperolet — APPROVED** [review]
:+1:

**Pull request merged**